api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -313,8 +312,8 @@ private void checkAndAddPartitionName(String name) {
       partitionNames.add(name);
     }
 
-    public Builder withSpecId(int specId) {
-      this.specId = specId;
+    public Builder withSpecId(int newSpecId) {
```

Comment:
What are your thoughts on disabling this check? Seems like this pattern is used throughout the code and we'd have to come up with creating names for the parameters, even though `this.name = name` provides good enough disambiguation

Reply:
Parameters cannot hide field names.

Reply:
Again we run into the same problem of the majority of cases being better to follow the standard - generally naming vars and parameters the same as fields is dubious. I think it's ok that in some cases where it might be considered acceptable to break this trend, to nevertheless conform to the global standard. Otherwise, we'd have to add a `SuppressWarnings` here.

Reply:
I'd rather use `newSpecId` than `SuppressWarnings` for cases like this. But, I agree that this makes things more awkward. I wish we could allow cases where the field is assigned to an argument with the same name.

Reply:
This rule will not apply to constructors, right?